src/finder.rs

```diff
@@ -17,6 +17,20 @@ pub struct Link<'t> {
     kind: LinkKind,
 }
 
+/// The type of trigger used for scanning the input
+/// Users should not exhaustively match this enum, because more triggers
+/// may be added in the future.
+#[derive(Debug, PartialEq)]
+pub enum Trigger {
```

Comment:
Can you make this `pub(crate)`? Want to make it clear at this point that it's not part of the API.